raiden/storage/sqlite.py

```diff
@@ -292,6 +307,89 @@ def get_latest_state_change_by_data_field(
 
         return result
 
+    def _get_state_changes(
+            self,
+            limit: int = None,
+            offset: int = None,
+            filters: Dict[str, Any] = None,
+    ) -> List[StateChangeRecord]:
+        """ Return a batch of state change records (identifier and data)
+
+        The batch size can be tweaked with the `limit` and `offset` arguments.
+
+        Additionally the returned state changes can be optionally filtered with
+        the `filters` parameter to search for specific data in the state change data.
+        """
+        limit, offset = _sanitize_limit_and_offset(limit, offset)
+        cursor = self.conn.cursor()
+
+        query = 'SELECT identifier, data FROM state_changes '
+
+        where_clauses = []
+        args = []
+        if filters:
+            for field, value in filters.items():
+                where_clauses.append('json_extract(data, ?)=?')
+                args.append(f'$.{field}')
+                args.append(value)
+
+            query += (
+                f"WHERE {' AND '.join(where_clauses)}"
+            )
+
+        query += 'ORDER BY identifier ASC LIMIT ? OFFSET ?'
+        args.append(limit)
+        args.append(offset)
+
+        cursor.execute(query, args)
+        result = []
+        try:
+            rows = cursor.fetchall()
+            for row in rows:
```

Comment:
This code can be greatly simplified:

```python
result = [
   StateChangeRecord(row[0], row[1])
   # cursor is an iterable so .fetchall() is not required, it just do an extra intermediary allocation
   for row in cursor
]
```

Reply:
Thank you. That's a good tip. Changed it here and in some other parts of sqlite.py with https://github.com/raiden-network/raiden/pull/3550/commits/73bb4205f6e611094beec393219463c0dd3dbb9c